Tests/SQLKitTests/SQLKitTests.swift

```diff
@@ -717,4 +717,41 @@ CREATE TABLE `planets`(`id` BIGINT, `name` TEXT, `diameter` INTEGER, `galaxy_nam
             XCTFail("Could not decode row with keyDecodingStrategy \(error)")
         }
     }
+
+    func testUnion() throws {
+        try db.union {
+            $0.column("id")
+                .from("t1")
+                .where("f1", .equal, "foo")
+                .limit(1)
+        }.union({
+            $0.column("id")
+                .from("t2")
+                .where("f2", .equal, "bar")
+                .limit(2)
+        }).union({
+            $0.column("id")
+                .from("t3")
+                .where("f3", .equal, "baz")
+                .limit(3)
+        })
```

Comment:
Note that

```swift
db.union {
}
.union {
}
...
```

fails with `Ambiguous use of 'union'` in the first line. Subsequent closures need to be placed in `(...)`.

Reply:
This can be somewhat mitigated by adding

```swift
extension SQLSelectBuilder {
    public func union(distinct predicate: (SQLSelectBuilder) -> SQLSelectBuilder) -> SQLUnionBuilder {
        return SQLUnionBuilder(on: self.database, initialQuery: self.select)
            .union(distinct: predicate(.init(on: self.database)).select)
    }

    /// Alias the `distinct` variant so it acts as the "default".
    public func union(_ predicate: (SQLSelectBuilder) -> SQLSelectBuilder) -> SQLUnionBuilder {
        return SQLUnionBuilder(on: self.database, initialQuery: self.select)
            .union(distinct: predicate(.init(on: self.database)).select)
    }

    public func union(all predicate: (SQLSelectBuilder) -> SQLSelectBuilder) -> SQLUnionBuilder {
        return SQLUnionBuilder(on: self.database, initialQuery: self.select)
            .union(all: predicate(.init(on: self.database)).select)
    }
}
```

allowing:

```swift
db.select()
            .column("id")
            .from("t1")
            .where("f1", .equal, "foo")
            .limit(1)
            .union({
                $0.column("id")
                    .from("t2")
                    .where("f2", .equal, "bar")
                    .limit(2)
            }).union(all: {
                $0.column("id")
                    .from("t3")
                    .where("f3", .equal, "baz")
                    .limit(3)
            })
```

I'll commit that as well now.

Overall, I think the closure based syntax is a little clunkier in use but on the other hand it allows expressing `... UNION ... UNION ALL ...`. I don't think that's possible with my initially proposed API. (You can probably nest queries to work around that though.)

The biggest problem for me is the implementation. I understand maybe 64% of it today, and that understanding will decay rapidly for here. I hope it's going to make sense for future maintainers 😅

Reply:
@gwynne , @0xTim - just checking - are there any further changes required to get this merged? :)